targets/unity-v2/source/Authentication/PlayFabAuthService.cs

```diff
@@ -0,0 +1,228 @@
+            AuthStrategies.Add(AuthTypes.Twitch, new TwitchAuthStrategy());
+            AuthStrategies.Add(AuthTypes.Steam, new SteamAuthStrategy());
+        }
+
+        public bool IsLoggedIn()
+        {
+            return AuthenticationContext != null;
+        }
+
+        public void ClearRememberMe()
+        {
+            PlayerPrefs.DeleteKey(_LoginRememberKey);
+            PlayerPrefs.DeleteKey(_PlayFabRememberMeIdKey);
+        }
+
+        /// <summary>
+        /// Kick off the authentication process by specific authtype.
+        /// </summary>
+        /// <param name="authType"></param>
+        public void Authenticate(AuthTypes authType)
+        {
+            AuthType = authType;
+            Authenticate();
+        }
+
+        public void Authenticate()
+        {
+            var authType = AuthType;
+
+            if (authType == AuthTypes.None)
+            {
+                if(OnDisplayAuthentication != null)
+                    OnDisplayAuthentication.Invoke();
+                return;
+            }
+
+            var auth = AuthStrategies[authType];
+            if (auth == null)
+            {
+                Debug.LogError("Unhandled auth type: " + authType);
+                return;
+            }
+
+            auth.Authenticate(this, resultCallback =>
+            {
+                // Store Identity and session
+                AuthenticationContext = resultCallback.AuthenticationContext;
+                PlayFabId = resultCallback.PlayFabId;
+                SessionTicket = resultCallback.SessionTicket;
+
+                EntityToken = new EntityKey
+                {
+                    Id = resultCallback.EntityToken.Entity.Id,
+                    Type = resultCallback.EntityToken.Entity.Type
+                };
+
+                // check if we want to get this callback directly or send to event subscribers.
+                if (OnLoginSuccess != null)
+                    OnLoginSuccess.Invoke(resultCallback); // report login result back to the subscriber
+            }, errorCallback =>
+            {
+                // report error back to the subscriber
+                if (OnPlayFabError != null)
+                    OnPlayFabError.Invoke(errorCallback);
+                else Debug.LogError(errorCallback.GenerateErrorReport());
+            });
+        }
+
+        internal void InvokeLink(AuthTypes linkType, PlayFabError error = null)
+        {
+            if(OnPlayFabLink != null)
+                OnPlayFabLink.Invoke(linkType, error);
+        }
+
+        internal void InvokeUnlink(AuthTypes unlinkType, PlayFabError error = null)
+        {
+            if(OnPlayFabUnlink != null)
+                OnPlayFabUnlink.Invoke(unlinkType, error);
+        }
+
+        internal void InvokeDisplayAuthentication()
+        {
+            if (OnDisplayAuthentication != null)
+                OnDisplayAuthentication.Invoke();
+        }
+
+        public void Link(AuthTypes linkType)
```

Comment:
This method reinforces my concern with calling Link(FB) and then Link(Google), and AuthToken being a shared value.
In fact, it tips the scales. It's not an @MarcoWilliamsPF question anymore. We must delete the shared AuthToken value, and break out separate inputs for every authentication type.
Otherwise we can't support two simultaneous recoverable logins.